### ISSUE TITLE
Error compiling on Mac OS Big Sur

### ISSUE THREAD

**Issue by glopezep** [rathena/rathena#6591]
```
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:1848:38: error: implicit instantiation of undefined template 'std::__1::hash<e_elemental_skillmode>'
    : public integral_constant<bool, __is_empty(_Tp)> {};
                                     ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/unordered_map:427:18: note: in instantiation of template class 'std::__1::is_empty<std::__1::hash<e_elemental_skillmode>>' requested here
          bool = is_empty<_Hash>::value && !__libcpp_is_final<_Hash>::value>
                 ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/unordered_map:862:13: note: in instantiation of default argument for '__unordered_map_hasher<e_elemental_skillmode, std::__1::__hash_value_type<e_elemental_skillmode, std::__1::shared_ptr<s_elemental_skill>>, std::__1::hash<e_elemental_skillmode>>' required here
    typedef __unordered_map_hasher<key_type, __value_type, hasher>   __hasher;
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./elemental.hpp:64:80: note: in instantiation of template class 'std::__1::unordered_map<e_elemental_skillmode, std::__1::shared_ptr<s_elemental_skill>>' requested here
        std::unordered_map<e_elemental_skillmode, std::shared_ptr<s_elemental_skill>> skill;    /// mode, skill
                                                                                      ^
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/../include/c++/v1/type_traits:428:50: note: template is declared here
template <class _Tp> struct _LIBCPP_TEMPLATE_VIS hash;
                                                 ^
1 error generated.
make[1]: *** [obj/atcommand.o] Error 1
make: *** [map] Error 2
```

**Comment by aleos89** [comment]
Can you do `#include <unordered_map>` at the top of your elemental.hpp and let me know the results?

**Comment by merryhime** [comment]
This is a known deficit in the C++11 standard which is [resolved in C++14](https://cplusplus.github.io/LWG/issue2148). Libraries have supported `std::hash` of enum types prior but this is technically non-standard behaviour. See above PR.

**Comment by Lemongrass3110** [comment]
Fixed in 31c7e4b
